tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -171,87 +153,30 @@ def exponential_delay(base_delay: int, exponential_backoff: int = 2) -> int:
         None
     """
     try:
-        _base_delay = int(base_delay)
-        _exponential_backoff = int(exponential_backoff)
-        delay = _base_delay + (random.randint(0, 1000) / 1000.0)  # nosec
-        LOGGER.debug(f"Performing back off retry sleeping {delay} seconds")
-        time.sleep(delay)
-        return _base_delay * _exponential_backoff
+        int(base_delay)
```

Comment:
Alright. I did remove the second int cast on base_delay.

Reply:
Went with storing the cast so we don't have to cast on each usage.

Reply:
Maybe... The abundance of int casts gives me pause. How about this? Keeps the error gates and business logic separate, but fully handles casting without weird side-cases when base_delay is a string containing numbers.

Reply:
It also allows strings through, whereupon time.sleep will raise an error. And if we're not worried about strings, then we shouldn't be doing these checks anyway.

Reply:
Actually if you did a string you would get a TypeError

```
>>> import random
>>>
>>> random_addition = random.randint(0, 1000) / 1000.0
>>>
>>> base_delay = "x"
>>> exponential_backoff = "y"
>>>
>>> try:
...     delay = base_delay + random_addition
...     new_backoff_delay = base_delay + exponential_backoff
... except TypeError as te:
...     raise te
...
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "<stdin>", line 2, in <module>
TypeError: can only concatenate str (not "float") to str
```

Reply:
Ah, we get saved by exponential_backoff being a float. This would mean that we'd lose the descriptive error message "ValueError: invalid literal for int() with base 10: 'x'".